packages/issuer-token-endpoint/lib/IssuerTokenEndpoint.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  AccessTokenResponse,
+  Alg,
+  CNonceState,
+  CredentialOfferState,
+  getNumberOrUndefined,
+  IStateManager,
+  Jwt,
+  JWTSignerCallback,
+  Typ,
+} from '@sphereon/openid4vci-common'
+import express, { NextFunction, Request, Response, Router } from 'express'
+import { v4 } from 'uuid'
+
+const router = express.Router()
+
+interface ITokenEndpointOpts {
+  tokenPath?: string
+  interval?: number
+  cNonceExpiresIn?: number
+  tokenExpiresIn?: number
+  stateManager?: IStateManager<CredentialOfferState>
+  nonceStateManager?: IStateManager<CNonceState>
+  jwtSignerCallback?: JWTSignerCallback
+}
+
+let tokenPath: string
+let interval: number
+let cNonceExpiresIn: number
+let tokenExpiresIn: number
+let stateManager: IStateManager<CredentialOfferState>
+let nonceStateManager: IStateManager<CNonceState>
+let jwtSignerCallback: JWTSignerCallback
+
+export const tokenRequestEndpoint = (opts?: ITokenEndpointOpts): Router => {
+  tokenPath = (opts?.tokenPath ? opts.tokenPath : (process.env.TOKEN_PATH as string)) ?? '/token'
+  interval = (opts?.interval ? opts.interval : getNumberOrUndefined(process.env.INTERVAL)) ?? 300000
+  cNonceExpiresIn = (opts?.cNonceExpiresIn ? opts.cNonceExpiresIn : getNumberOrUndefined(process.env.C_NONCE_EXPIRES_IN)) ?? 300000
+  tokenExpiresIn = (opts?.tokenExpiresIn ? opts.tokenExpiresIn : getNumberOrUndefined(process.env.TOKEN_EXPIRES_IN)) ?? 300000
+  if (opts?.jwtSignerCallback) {
+    jwtSignerCallback = opts.jwtSignerCallback
+  } else {
+    throw new Error('Please provide a private a JWT signer function')
+  }
+  if (opts?.stateManager) {
+    stateManager = opts.stateManager
+  } else {
+    throw new Error('Unable to proceed without an StateManager instance')
+  }
+  if (opts.nonceStateManager) {
+    nonceStateManager = opts.nonceStateManager
+  } else {
+    throw new Error('Unable to proceed without an NonceStateManager instance')
+  }
+  router.post(tokenPath, handleHTTPStatus400, handleTokenRequest)
+  return router
+}
+
+const generateAccessToken = async (): Promise<string> => {
+  const issuanceTime = new Date()
+  const jwt: Jwt = {
+    header: { typ: Typ.JWT, alg: Alg.ES256 },
+    payload: { iat: issuanceTime.getTime(), exp: +new Date(issuanceTime.getTime() + tokenExpiresIn) },
```

Comment:
I would expect a relation to the actual credential offer somehow in the access token, so we can get back to the state. This is a JWT signed by us, so it makes sense that we will trust it coming back in as long as the signature is valid and from us

Reply:
I'm binding the state to the iss property. I think it's what makes more sense.